moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -957,14 +957,14 @@ bool TrajectoryExecutionManager::validate(const TrajectoryExecutionContext& cont
       // Check single-dof trajectory
       const std::vector<double>& positions = trajectory.joint_trajectory.points.front().positions;
       const std::vector<std::string>& joint_names = trajectory.joint_trajectory.joint_names;
-      const std::size_t n = joint_names.size();
-      if (positions.size() != n)
+      if (positions.size() != joint_names.size())
       {
-        ROS_ERROR_NAMED("traj_execution", "Wrong trajectory: #joints: %zu != #positions: %zu", n, positions.size());
+        ROS_ERROR_NAMED("traj_execution", "Wrong trajectory: #joints: %zu != #positions: %zu", joint_names.size(),
+                        positions.size());
         return false;
       }
 
-      for (std::size_t i = 0; i < n; ++i)
+      for (std::size_t i = 0, end = joint_names.size(); i < end; ++i)
```

Comment:
Where is ``end`` defined? Is this some C++ magic?

Reply:
Just right after `i=0` in the for loop's init condition. No magic there ;-)